### PR TITLE
[SW-2092] Switch H2OFrameSupport methods to use rest api

### PR DESCRIPTION
**Pull request by jakubhava**
JIRA for removal in 3.32 is here https://0xdata.atlassian.net/browse/SW-2134

### CODE REVIEW

File: core/src/main/scala/ai/h2o/sparkling/H2OFrame.scala [hunk 1]
```diff
@@ -47,37 +47,51 @@ class H2OFrame private(val frameId: String, val columns: Array[H2OColumn], priva
...
       )
       val rapidsFrameV3 = update[RapidsFrameV3](endpoint, "99/Rapids", conf, params)
       H2OFrame(rapidsFrameV3.key.name)
     }
   }
 
-  def splitToTrainAndValidationFrames(splitRatio: Double): Array[H2OFrame] = {
-    if (splitRatio >= 1.0) {
-      throw new IllegalArgumentException("Split ratio must be lower than 1.0")
+  def split(splitRatios: Array[Double]): Array[H2OFrame] = {
```
Comment (jakubhava) [inline core/src/main/scala/ai/h2o/sparkling/H2OFrame.scala]: I think it is better to keep the name just to `split`. We can do this change without any deprecation as this API has not been announced so far

File: core/src/main/scala/ai/h2o/sparkling/backend/utils/H2OClientUtils.scala [hunk 2]
```diff
@@ -140,4 +141,23 @@ object H2OClientUtils extends SharedBackendUtils {
...
+  /**
+   * This method should be used whenever the Frame needs to be updated. This method ensures to use proper
+   * locking mechanism.
+   *
+   * @param fr frame to update
+   * @param f  function to run on the frame
+   * @tparam T H2O Frame Type
+   * @return returns the updated frame
+   */
+  def withLockAndUpdate[T <: Frame](fr: T)(f: T => Any): T = {
```
Comment (jakubhava) [inline core/src/main/scala/ai/h2o/sparkling/backend/utils/H2OClientUtils.scala]: Keep the few specific client related methods in this object so they are not scattered around the code

File: core/src/main/scala/ai/h2o/sparkling/H2OFrame.scala [hunk 3]
```diff
@@ -47,37 +47,52 @@ class H2OFrame private(val frameId: String, val columns: Array[H2OColumn], priva
...
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/H2OFrame.scala]: nit: Isn't better have just one method with multiple number of parameter?

Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/H2OFrame.scala]: yup, will do

**Pull request merged**